student.cpp

```diff
@@ -206,28 +209,44 @@ float Student::GradeablePercent(GRADEABLE_ENUM g) const {
 
   // collect the scores in a vector
   std::vector<score_object> scores;
+  std::vector<score_object> only_ec;
+
   for (int i = 0; i < GRADEABLES[g].getCount(); i++) {
     float s = getGradeableItemGrade(g,i).getValue();
     std::string id = GRADEABLES[g].getID(i);
     float m = nonzero_sum/nonzero_count;
     //if(!id.empty() && GRADEABLES[g].isReleased(id)){
     if(!id.empty()){
       m = GRADEABLES[g].getItemMaximum(id);
-      // std::cout << "m" << m << std::endl;
+//       std::cout << "m" << m << std::endl;
     }
+    bool ec = GRADEABLES[g].isExtraCredit(id);
     float p = GRADEABLES[g].getItemPercentage(id);
     float sm = GRADEABLES[g].getScaleMaximum(id);
-    scores.push_back(score_object(s,m,p,sm));
+    if(!ec) {
+        scores.push_back(score_object(s,m,p,sm,ec));
+    } else {
+        only_ec.push_back(score_object(s,m,p,sm,ec));
+    }
   }
 
   // sort the scores (smallest first)
+
   std::sort(scores.begin(),scores.end());
   //to check that the number of "drop the lowest" is less than the number of non extra credit gradeables,
   // i.e., it is not allowed to drop all non extra credit gradeables
   assert (GRADEABLES[g].getRemoveLowest() >= 0 && (
           (non_extra_credit_count > 0 && GRADEABLES[g].getRemoveLowest() < non_extra_credit_count)) ||
           (GRADEABLES[g].getRemoveLowest() == 0));
 
+  if (!( ((GRADEABLES[g].getRemoveLowest() >= 0) && (
+            (non_extra_credit_count > 0 && GRADEABLES[g].getRemoveLowest() < non_extra_credit_count))) ||
+          (GRADEABLES[g].getRemoveLowest() == 0))) {
+        std::cerr << "Error: Invalid value for removeLowest in gradeable Check the conditions.\n";
```

Comment:
The parameter is called "remove_lowest", so the error message should say ``std::cerr << "ERROR: Invalid value for remove_lowest in Rainbow Grades configuration. Check customization.json." << std::endl;``